naloge/poslovnez.tex

```diff
@@ -110,7 +110,56 @@
 \pgfslika[poslovnez-resitev2]
 \end{slika}
     
+\newpage
+\begin{small}
+\begin{algorithmic}
+\Function{MAKSIMALNIDOBIČEK}{$G = (V, E), s \in V, L : E \to \R$}
+    \State $d \gets$ seznam slovarjev z vrednostjo $- \infty$ za vsako vozlišče $v \in V$
+    \State $\pred \gets$ seznam slovarjev z vrednostjo $\Null$
+    za vsako vozlišče $v \in V$
+    \State $d[0][s] \gets 0$
+    \State $i \gets 0$
+    \State $\text{\sl trenutna} \gets \{s\}$
+    \While{$\lnot \text{\sl trenutna}.\isEmpty()$}
+        \State $i \gets i+1$
+        \If{$i > |V|$} \hfill $\exists$ pozitiven cikel
+            \State \Return graf ima negativen cikel 
+        \EndIf
+        \State $\text{\sl{naslednja}} \gets$ prazna množica
+        \For{$u \in \text{\sl trenutna}$}
+            \State $d[i][u] \gets d[i-1][u]$
+            \State $\pred[i][u] \gets u$
+        \EndFor
+        \For{$uv \in E$}
+            \If{$d[i][v] < d[i-1][u] +  L_{uv}$}
+                \State $d[i][v] \gets d[i-1][u] +  L_{uv}$
+                \State $\pred[i][v] \gets u$
+            \EndIf
+        \EndFor
+        \State $\text{\sl{trenutna}} \gets \text{\sl{naslednja}}$
+    \EndWhile
+    \State $zasluzek \gets$ max $({d_{i,u} |i=0, \ldots , |V|;  u \in V})$
+    \State $konec \gets \{v | d_{i,v}= zasluzek \}$
```

Comment:
Tole bi se lepše zapisalo tako:
```LaTeX
    \State $(\text{\sl zaslužek}, \text{\sl konec}) \gets \max\{(d[i][u], u) \mid i = 0, 1, \dots, |V|, \ u \in V\}$
```
Morda pa bo še bolje, če si kar sproti (vsakič, ko poračunaš novo razdaljo) beležiš največjo razdaljo ter ustrezno vozlišče in globino (`i`), tako da ti jih ni potrebno iskati tukaj.